src/components/modal/types.ts

```diff
@@ -1,10 +1,33 @@
 import { ForwardRefExoticComponent, HTMLAttributes, RefAttributes } from 'react'
 
+// Define an enum-like type for the possible variants of the modal
+export type ModalVariant = 'small' | 'medium'
 export interface ModalProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * Indicates whether the modal is open or closed
+   */
   isOpen: boolean
+  /**
+   * Callback function to be called when the `ModalBg` has been clicked
+   * or when the user presses the `Escape` key
+   */
   onModalClose: () => void
+
+  /**
+   * The title of the modal
+   */
   title?: string
-  className?: string
+
+  /**
+   * The variant of the modal (small or medium)
+   * @defaultValue 'medium'
+   */
+  variant?: ModalVariant
+
+  /**
+   * Optional content that will placed under `ModalFooter`
+   */
+  footer?: React.ReactNode
```

Comment:
raise a query to Design team, since I'm also not sure should we make it required or not for the footer content, and curious about the allowed component for dialog footer 

https://www.figma.com/design/J6EneNGfpk2GcUuymiOKW4?node-id=164-4207&m=dev#978864433